### ISSUE TITLE
Compiling third-party dependencies does not work correctly

### ISSUE THREAD

**Issue by syinari0123** [jczarnowski/DeepFactors#12]
Thank you for publishing such an excellent code!

I tried to compile it with running `makedeps.sh`, I got the following error:
```
./thirdparty/makedeps.sh: line 148: patch/brisk-missing-functional.patch: No such file or directory
./thirdparty/makedeps.sh: line 149: patch/brisk-opencv4.patch: No such file or directory
```

It appears that downloaded brisk (version 2.0.5) was not unzipped with the correct directory structure.
I eventually resolved this by changing the version of brisk from 2.0.5 to 2.0.2!

It is recommended that you change the version of brisk in `makedeps.sh` to 2.0.2 or change the code to allow download as an appropriate directory structure.

**Comment by jczarnowski** [comment]
Hi, thanks for your kind words!

Based on the error you have copied, the script seems to complain that it can't find the two patch files, but they are included in our repository and are downloaded during git clone. I am a bit confused on how they went missing in your case.

**Comment by jczarnowski** [comment]
I've checked things on my side and I can confirm that I am also affected by the brisk unpacking issue. I'll investigate further

**Comment by jczarnowski** [comment]
I am not entirely sure why I have not hit this issue before, but the linked pull request should fix it without downgrading to Brisk 2.0.2. In short, the structure of the ZIP file has changed between 2.0.2 and 2.0.5 and I've adapted the unzipping method to simply wrap the files in a directory.

**Comment by syinari0123** [comment]
Thank you for your quick response!
